data/crac/crac-impl/src/test/java/com/farao_community/farao/data/crac_impl/InjectionSetpointAdderImplTest.java

```diff
@@ -54,14 +56,28 @@ void testOk() {
     @Test
     void testNoNetworkElement() {
         InjectionSetpointAdder injectionSetpointAdder = networkActionAdder.newInjectionSetPoint()
-            .withSetpoint(100.);
+            .withSetpoint(100.).withUnit(Unit.MEGAWATT);
         assertThrows(FaraoException.class, injectionSetpointAdder::add);
     }
 
     @Test
     void testNoSetpoint() {
         InjectionSetpointAdder injectionSetpointAdder = networkActionAdder.newInjectionSetPoint()
-            .withNetworkElement("groupNetworkElementId");
+            .withNetworkElement("groupNetworkElementId").withUnit(Unit.MEGAWATT);
         assertThrows(FaraoException.class, injectionSetpointAdder::add);
     }
+
+    @Test
+    void testNoUnit() {
+        InjectionSetpointAdder injectionSetpointAdder = networkActionAdder.newInjectionSetPoint()
+                .withNetworkElement("groupNetworkElementId").withSetpoint(100.);
+        assertThrows(FaraoException.class, injectionSetpointAdder::add);
+    }
+
+    @Test
+    void testNegativeSetPointWithSectionCount() {
+        InjectionSetpointAdder injectionSetpointAdder = networkActionAdder.newInjectionSetPoint()
+                .withNetworkElement("groupNetworkElementId").withSetpoint(-100.);
+        assertThrows(FaraoException.class, () -> injectionSetpointAdder.withUnit(Unit.SECTION_COUNT));
```

Comment:
- check exception message
- also test that add() fails if setpoint is not an integer in a new test (for example setting setpoint to 1.3)

Reply:
Done !